portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -0,0 +1,552 @@
+
+        sqlService.dbInsert(null
+            , BULLHORN_INSERT_SQL
+            , new Object[] {ACADEMIC, from, to, event, ref, title, siteId, eventTime, url}
+            , "ID");
+    }
+
+    private void doSocialInsert(String from, String to, String event, String ref, Date eventTime ,String url) {
+
+        sqlService.dbInsert(null
+            , BULLHORN_INSERT_SQL
+            , new Object[] {SOCIAL, from, to, event, ref, "", "", eventTime, url}
+            , "ID");
+    }
+
+    private void sendCommentAlerts(String from, String event, String ref, Event e, String siteId, String postId, Cache countCache, Set<String> tos) {
+
+        log.debug("sending comment alerts: from is {}, tos is {}, siteId is {}", from, tos, siteId);
+        boolean isSocial = siteId.equals("SOCIAL");
+        for (String to : tos) {
+            // If we're commenting on our own post, no alert needed
+            if (!from.equals(to)) {
+                String mySiteId = siteId;
+                if (isSocial) {
+                    mySiteId = "~" + to;
+                }
+                Site site = null;
+                try {
+                    site = siteService.getSite(mySiteId);
+                } catch (IdUnusedException ex) {
+                    log.error("Couldn't find site " + mySiteId, ex);
+                }
+                if (site != null) {
+                    String toolId = site.getToolForCommonId("sakai.commons").getId();
+                    String url = serverConfigurationService.getPortalUrl() + "/directtool/"
+                                 + toolId + "/posts/" + postId;
+                    if (isSocial) {
+                        doSocialInsert(from, to, event, ref, e.getEventTime(), url);
+                    } else {
+                        String title = "";
+                        doAcademicInsert(from, to, event, ref, title, siteId, e.getEventTime(), url);
+                    }
+                    countCache.remove(to);
+                }
+            }
+        }
+    }
+
+    public List<BullhornAlert> getSocialAlerts(String userId) {
+
+        List<BullhornAlert> alerts = sqlService.dbRead(ALERTS_SELECT_SQL
+                , new Object[] { SOCIAL, userId }
+                , new SqlReader() {
+                        public Object readSqlResultRecord(ResultSet rs) {
+                            return new BullhornAlert(rs);
+                        }
+                    }
+                );
+
+        for (BullhornAlert alert : alerts) {
+            try {
+                User fromUser = userDirectoryService.getUser(alert.from);
+                alert.fromDisplayName = fromUser.getDisplayName();
+            } catch (UserNotDefinedException unde) {
+                alert.fromDisplayName = alert.from;
+            }
+        }
+
+        return alerts;
+    }
+
+    public int getSocialAlertCount(String userId) {
+
+        Cache countCache = memoryService.newCache("bullhorn_alert_count_cache");
```

Comment:
Yep, no probs. I only did that because the MemoryService.newCache pulls the existing one if it has been created already and I thought the overhead minimal with the benefit of the provenance being obvious. Actually, newCache is deprecated, so I shouldn't be using that anyway ...